src/views/forms/AccountManagementFormsNew/SlippageEditor.tsx

```diff
@@ -0,0 +1,150 @@
+import { useEffect, useRef, useState, type Dispatch, type SetStateAction } from 'react';
+
+import type { NumberFormatValues } from 'react-number-format';
+import styled from 'styled-components';
+
+import { ButtonShape, ButtonSize } from '@/constants/buttons';
+import { STRING_KEYS } from '@/constants/localization';
+
+import { useStringGetter } from '@/hooks/useStringGetter';
+
+import { FormInput } from '@/components/FormInput';
+import { Icon, IconName } from '@/components/Icon';
+import { InputType } from '@/components/Input';
+import { Output, OutputType } from '@/components/Output';
+import { ToggleGroup } from '@/components/ToggleGroup';
+import { WithConfirmationPopover } from '@/components/WithConfirmationPopover';
+
+enum EditorState {
+  Viewing = 'Viewing',
+  Selecting = 'Selecting',
+  Editing = 'Editing',
+}
+
+type ElementProps = {
+  slippage: number;
+  setIsEditing?: Dispatch<SetStateAction<boolean>>;
+  setSlippage: (slippage: number) => void;
+  disabled?: boolean;
+};
+
+export type SlippageEditorProps = ElementProps;
+
+export const SlippageEditor = ({
+  disabled,
+  slippage,
+  setIsEditing,
+  setSlippage,
+}: SlippageEditorProps) => {
+  const percentSlippage = slippage * 100;
+  const [slippageInputValue, setSlippageInputValue] = useState(percentSlippage.toString());
+  const [editorState, setEditorState] = useState(EditorState.Viewing);
+
+  const stringGetter = useStringGetter();
+
+  const slippageOptionItems = [
+    { label: '0.1%', value: '0.001' },
+    { label: '0.25%', value: '0.0025' },
+    { label: stringGetter({ key: STRING_KEYS.CUSTOM }), value: 'custom' },
+  ];
+  const toggleGroupRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    setIsEditing?.(editorState !== EditorState.Viewing);
```

Comment:
i think it's strange that this child component needs to set parent state as a byproduct here - i prefer we just remove the `isEditingSlippage` requirement from `DepositButtonAndReceipt` since that doesn't feel necessary